incubator/hnc/pkg/controllers/object_controller.go

```diff
@@ -284,14 +291,31 @@ func (r *ObjectReconciler) onDelete(ctx context.Context, log logr.Logger, nnm ty
 	return nil
 }
 
+func (r *ObjectReconciler) ensureCanPropagate(nsNm string, inst *unstructured.Unstructured) bool {
+	r.Forest.Lock()
+	defer r.Forest.Unlock()
+	ns := r.Forest.Get(nsNm)
+	if !ns.Exists() {
+		return false
+	}
+	if inst.GetDeletionGracePeriodSeconds() != nil {
+		ns.SetCondition(fmt.Sprint(r.GVK, ns, nsNm), tenancy.ObjectInternal, "finalizer set")
```

Comment:
A few things here:

- The key needs to be separated by slashes (`group/version/kind/namespace/name`), does your code do this? My reading of `fmt.Sprint` is that is separates the args using spaces.
- Not `"finalizer set"` - more like `"object is being deleted (deletionGracePeriodSeconds is set)"`

Reply:
It does not. Again, should have asked when I thought of it.